numpy/linalg/tests/test_linalg.py

```diff
@@ -882,9 +900,12 @@ def do(self, a, b, tags):
             expect_resids = np.array([]).view(type(x))
         assert_almost_equal(residuals, expect_resids)
         assert_(np.issubdtype(residuals.dtype, np.floating))
-        assert_(imply(isinstance(b, matrix), isinstance(x, matrix)))
-        assert_(imply(isinstance(b, matrix), isinstance(residuals, matrix)))
+        expected_type = type(b) if isinstance(b, np.ndarray) else np.ndarray
```

Comment:
`type(b) if isinstance(b, np.ndarray) else np.ndarray` is a clear candidate for a helper function somewhere in this file.

Reply:
OK, done